src/Swift.Runtime/src/Swift/Set.cs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Swift;
+using Swift.Runtime;
+using Swift.Runtime.InteropServices;
+
+namespace Swift;
+
+/// <summary>
+/// Represents a Swift set payload.
+///
+/// The following diagram illustrates the hierarchy of a Swift set type.
+/// The actual implementation may differ:
+///
+///    struct Set
+///    +-----------------------------------------------------------------------+
+///    |   struct Variant                                                     |
+///    |   +--------------------------------------------------------------+   |
+///    |   |   struct Object                                              |   |
+///    |   |   +------------------------------------------------------+   |   |
+///    |   |   | var rawValue: IntPtr                                 |   |   |
+///    |   |   +------------------------------------------------------+   |   |
+///    |   +--------------------------------------------------------------+   |
+///    +-----------------------------------------------------------------------+
+/// </summary>
+[StructLayout(LayoutKind.Sequential, Size = 8)]
+public struct Variant
+{
+    public IntPtr rawValue;
+}
+
+/// <summary>
+/// Represents a Swift set.
+/// </summary>
+/// <typeparam name="Element">The element type contained in the set.</typeparam>
+public class SwiftSet<Element> : IDisposable, ISwiftObject
+{
+    static nuint _payloadSize = SwiftObjectHelper<SwiftSet<Element>>.GetTypeMetadata().Size;
+
+    static nuint _elementSize = ElementTypeMetadata.Size;
+
+    private Variant _variant;
+
+    public unsafe void Dispose()
+    {
+        if (_variant.rawValue != IntPtr.Zero)
+        {
+            Arc.Release(*(IntPtr*)_variant.rawValue);
+            _variant.rawValue = IntPtr.Zero;
+            GC.SuppressFinalize(this);
+        }
+    }
+
+    unsafe ~SwiftSet()
+    {
+        Arc.Release(*(IntPtr*)_variant.rawValue);
+        _variant.rawValue = IntPtr.Zero;
+    }
```

Comment:
Is there a reason this doesn't use the canonical Dispose pattern (having a Dispose(bool disposing) method which is called from both dispose and finalizer?

Reply:
Added a tracking issue: https://github.com/dotnet/runtimelab/issues/2978. We should review the generated bindings as well.